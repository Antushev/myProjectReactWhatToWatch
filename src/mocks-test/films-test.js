import {generateFilm, generateFilms} from '../mocks/films.js';

const FILMS_NUMBER = 10;

const film = generateFilm(1);
const films = generateFilms(FILMS_NUMBER);

export {film, films};
