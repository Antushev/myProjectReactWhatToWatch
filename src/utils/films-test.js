const films = [
  {
    id: 1,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2001
  },
  {
    id: 2,
    name: `Интерстеллар`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2000
  },
  {
    id: 3,
    name: `Спасти рядоого Райана`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2004
  },
  {
    id: 4,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 5,
    name: `Власталин колец: Возвращение короля`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2014
  },
  {
    id: 6,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2019
  },
  {
    id: 7,
    name: `Хоббит: Нежданное путешествие`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2020
  },
  {
    id: 8,
    name: `Хоббит: Пустошь Смауга`,
    picture: `img/pulp-fiction.jpg`,
    genre: `Adventure`,
    date: 2018
  }
];

const film = {
  id: 1,
  name: `Интерстеллар`,
  picture: `img/pictrue.jpg`,
  genre: `Adventure`,
  date: 2014
};

export {film, films};
