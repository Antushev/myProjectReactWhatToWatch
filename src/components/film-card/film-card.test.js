import React from 'react';
import renderer from 'react-test-renderer';

import {film} from '../../mocks-test/films-test.js';

import FilmCard from './film-card.jsx';

describe(`FilmCardComponent`, () => {
  it(`FilmCardComponentSnapshot`, () => {
    const tree = renderer.create(
        <FilmCard
          film={film}
          handleFilmClick={() => {}}
          handleFilmCardMouseOver={() => {}}
          renderVideoPlayer={() => {}}
        />, {
          createNodeMock: () => {
            return {};
          }
        }
    )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
