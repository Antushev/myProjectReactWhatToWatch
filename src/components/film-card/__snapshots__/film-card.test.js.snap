// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmCardComponent FilmCardComponentSnapshot 1`] = `
<article
  className="small-movie-card catalog__movies-card"
>
  <a
    href="/films/1"
    onClick={[Function]}
  />
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="/films/1"
      onClick={[Function]}
    >
      Интерстеллар
    </a>
  </h3>
</article>
`;
